src/main/java/de/tum/cit/aet/artemis/core/security/ArtemisAuthenticationEventListener.java

```diff
@@ -0,0 +1,102 @@
+package de.tum.cit.aet.artemis.core.security;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.net.URL;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.authentication.event.AuthenticationSuccessEvent;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import de.tum.cit.aet.artemis.communication.service.notifications.MailSendingService;
+import de.tum.cit.aet.artemis.core.domain.User;
+import de.tum.cit.aet.artemis.core.exception.EntityNotFoundException;
+import de.tum.cit.aet.artemis.core.repository.UserRepository;
+
+/**
+ * Listener for successful authentication events in the Artemis system.
+ * This component listens for successful login attempts and sends email notifications
+ * to non-internal users when they successfully log in.
+ */
+@Profile(PROFILE_CORE)
+@Component
+public class ArtemisAuthenticationEventListener implements ApplicationListener<AuthenticationSuccessEvent> {
+
+    @Value("${artemis.user-management.password-reset.links.en:https://artemis.tum.de/account/reset/request}")
+    private String passwordResetLinkEnUrl;
+
+    @Value("${artemis.user-management.password-reset.links.de:https://artemis.tum.de/account/reset/request}")
+    private String passwordResetLinkDeUrl;
+
+    @Value("${server.url}")
+    private URL artemisServerUrl;
+
+    private static final Logger log = LoggerFactory.getLogger(ArtemisAuthenticationEventListener.class);
+
+    private final UserRepository userRepository;
+
+    private final MailSendingService mailSendingService;
+
+    public ArtemisAuthenticationEventListener(UserRepository userRepository, MailSendingService mailSendingService) {
+        this.userRepository = userRepository;
+        this.mailSendingService = mailSendingService;
+    }
+
+    /**
+     * Handles successful authentication events.
+     * Sends a login notification email to non-internal users when they successfully authenticate.
+     *
+     * @param event The authentication success event to process
+     */
+    @Override
+    public void onApplicationEvent(AuthenticationSuccessEvent event) {
```

Comment:
Is the javadoc accurate here? 

```suggestion
    /**
     * Handles successful authentication events.
     * Sends a login notification email to users when they successfully authenticate.
     *
     * @param event The authentication success event to process
     */
    @Override
    public void onApplicationEvent(AuthenticationSuccessEvent event) {
```